### PR TITLE
fix: show notifications for incoming MLS messages [WPB-7202]

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7202" title="WPB-7202" target="_blank">BugWPB-7202</a>  [iOS] Cannot get notifications for MLS conversations
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

MLS messages don't trigger push notifications.

### Causes (Optional)

The event type `.conversationMLSMessageAdd` is not handled.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Helpers/ZMUpdateEvent+Decryption.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ extension ZMUpdateEvent {
             return nil
         }
 
-        if self.type == .conversationOtrMessageAdd, let wrappedData = eventData["data"] as? String {
+        if self.type == .conversationOtrMessageAdd, let wrappedData = eventData["data"] as? String { // .conversationMLSMessageAdd?
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Helpers/ZMUpdateEvent+Decryption.swift]: not needed here, only proteus

File: wire-ios-request-strategy/Sources/Notifications/CallEventContent.swift [hunk 2]
```diff
@@ -64,7 +64,7 @@ public struct CallEventContent: Codable {
 
     public init?(from event: ZMUpdateEvent) {
         guard
-            event.type == .conversationOtrMessageAdd,
+            event.type.isOne(of: [.conversationOtrMessageAdd, .conversationMLSMessageAdd]),
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Notifications/CallEventContent.swift:67]: ✅

File: wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift [hunk 3]
```diff
@@ -26,7 +26,7 @@ public extension ZMLocalNotification {
         var builderType: EventNotificationBuilder.Type?
 
         switch event.type {
-        case .conversationOtrMessageAdd:
+        case .conversationOtrMessageAdd, .conversationMLSMessageAdd:
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/Content/ZMLocalNotification+Events.swift:29]: ✅

File: wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/LocalNotificationContentType.swift [hunk 4]
```diff
@@ -54,7 +54,7 @@ public enum LocalNotificationContentType: Equatable {
             let timeoutValue = MessageDestructionTimeoutValue(rawValue: TimeInterval(timeoutIntegerValueInSeconds))
             self = timeoutValue == .none ? .messageTimerUpdate(nil) : .messageTimerUpdate(timeoutValue.displayString)
 
-        case .conversationOtrMessageAdd:
+        case .conversationOtrMessageAdd: // .conversationMLSMessageAdd?
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Notifications/Push Notifications/Notification Types/LocalNotificationContentType.swift]: ✅

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift [hunk 5]
```diff
@@ -38,7 +38,7 @@ extension EventDecoder {
             return event
         }
 
-        guard event.type.isOne(of: .conversationOtrMessageAdd, .conversationOtrAssetAdd) else {
+        guard event.type.isOne(of: .conversationOtrMessageAdd, .conversationOtrAssetAdd) else { // .conversationMLSMessageAdd?
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift]: 🚫

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift [hunk 6]
```diff
@@ -279,7 +279,7 @@ private extension ZMUpdateEvent {
 
     var payloadKey: String? {
         switch type {
-        case .conversationOtrMessageAdd:
+        case .conversationOtrMessageAdd: // .conversationMLSMessageAdd?
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder+Proteus.swift]: 🚫

File: wire-ios-sync-engine/Source/Synchronization/Strategies/CallingRequestStrategy.swift [hunk 7]
```diff
@@ -247,7 +247,7 @@ public final class CallingRequestStrategy: AbstractRequestStrategy, ZMSingleRequ
 
     private func processEvent(_ event: ZMUpdateEvent) {
         let serverTimeDelta = managedObjectContext.serverTimeDelta
-        guard event.type == .conversationOtrMessageAdd || event.type == .conversationMLSMessageAdd else { return }
+        guard event.type.isOne(of: [.conversationOtrMessageAdd, .conversationMLSMessageAdd]) else { return }
```
Comment (David-Henner) [inline wire-ios-sync-engine/Source/Synchronization/Strategies/CallingRequestStrategy.swift:250]: ✅

File: wire-ios-sync-engine/Source/Synchronization/Strategies/TypingStrategy.swift
Comment (David-Henner) [inline wire-ios-sync-engine/Source/Synchronization/Strategies/TypingStrategy.swift:1]: ✅

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift
Comment (David-Henner) [inline wire-ios-transport/Source/Public/ZMUpdateEvent.swift:1]: @typfel or @johnxnguyen any thoughts on these?

File: wire-ios/Wire Notification Service Extension/Simple/Job.swift [hunk 8]
```diff
@@ -80,7 +80,7 @@ final class Job: NSObject, Loggable {
         let event = try await fetchEvent(eventID: eventID)
 
         switch event.type {
-        case .conversationOtrMessageAdd:
+        case .conversationOtrMessageAdd: // .conversationMLSMessageAdd ?
```
Comment (David-Henner) [inline wire-ios/Wire Notification Service Extension/Simple/Job.swift]: ✅

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift
Comment (typfel) [reply wire-ios-transport/Source/Public/ZMUpdateEvent.swift:1]: It make sense to add `.conversationMLSMessageAdd`. Mosts clients are probably announcing that they are no longer typing after they've sent a message but this a nice fail safe to consider a user not to be typing after receiving a text message from them.

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift [hunk 9]
```diff
@@ -224,7 +224,7 @@ open class ZMUpdateEvent: NSObject {
     /// True if the event contains cryptobox-encrypted data
     open var isEncrypted: Bool {
         switch self.type {
-        case .conversationOtrAssetAdd, .conversationOtrMessageAdd:
+        case .conversationOtrAssetAdd, .conversationOtrMessageAdd: // .conversationMLSMessageAdd?
```
Comment (typfel) [inline wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: Makes sense

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift [hunk 10]
```diff
@@ -234,7 +234,7 @@ open class ZMUpdateEvent: NSObject {
     /// True if the event is encoded with ZMGenericMessage
     open var isGenericMessageEvent: Bool {
         switch self.type {
-        case .conversationOtrMessageAdd, .conversationOtrAssetAdd, .conversationClientMessageAdd:
+        case .conversationOtrMessageAdd, .conversationOtrAssetAdd, .conversationClientMessageAdd: // .conversationMLSMessageAdd?
```
Comment (typfel) [inline wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: Makes sense

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift [hunk 11]
```diff
@@ -244,7 +244,7 @@ open class ZMUpdateEvent: NSObject {
     /// True if this event type could have two versions, encrypted and non-encrypted, during the transition phase
     open var hasEncryptedAndUnencryptedVersion: Bool {
         switch self.type {
-        case .conversationOtrMessageAdd,
+        case .conversationOtrMessageAdd, // .conversationMLSMessageAdd?
```
Comment (typfel) [inline wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: I guess it makes sense but we can delete this property, it was used during a transition phase when Wire introduced encryption.

File: wire-ios-transport/Source/Public/ZMUpdateEvent.swift (hunk 9 above)
Comment (typfel) [reply wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: Maybe this property is also not used anymore?

Comment (johnxnguyen) [reply wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: Is there an mls asset add event too?

Comment (caldrian) [reply wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: I see only those two:
```
case conversationMLSWelcome = 41
case conversationMLSMessageAdd = 42
```

Comment (caldrian) [reply wire-ios-transport/Source/Public/ZMUpdateEvent.swift]: I removed the property.

**Review by johnxnguyen — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**